### DISCUSSION TITLE
does builtin document has a string length limit?

### DISCUSSION THREAD

**Discussion by hyphappy011** [eclipse-langium/langium#1338] [Q&A] [answered]
when i built in my customer language doc which like typescript's declare files.  if the doc length greater than about 3200 lines, some doc content can not parse correctly.
The Code:
```ts
export class OneLangWorkspaceManager extends DefaultWorkspaceManager {

    private documentFactory: LangiumDocumentFactory;

    constructor(services: LangiumSharedServices) {
        super(services);
        this.documentFactory = services.workspace.LangiumDocumentFactory;
    }

    protected override async loadAdditionalDocuments(
        folders: WorkspaceFolder[],
        collector: (document: LangiumDocument<AstNode>) => void
    ): Promise<void> {
        await super.loadAdditionalDocuments(folders, collector);
        // Load our library using the `builtin` URI schema
        collector(this.documentFactory.fromString(oneLangBuiltin, URI.parse('builtin:///library.onelang')));
    }
}
```

**Comment by msujew** [comment, marked as answer]
Hey @hyphappy011,

there is no file length limit in Langium, neither for the builtin documents nor for workspace documents. It might be that there's a parser error in your document. Can you copy the content of your builtin library to the editor and see whether any errors appear in the document? Alternatively place a breakpoint in your workspace manager when loading the document. All documents track their lexer/parser errors in the `parseResult` property.

**Reply by hyphappy011** [reply]
Thank you @msujew
